src/EditorFeatures/VisualBasicTest/Recommendations/Statements/ElseKeywordRecommenderTests.vb

```diff
@@ -235,5 +235,50 @@ Select Case x
     Case 1
 </MethodBody>, "Else")
         End Sub
+
+        <Fact>
+        <Trait(Traits.Feature, Traits.Features.KeywordRecommending)>
+        Public Sub ElseWithinIfWithinElse()
+            VerifyRecommendationsContain(
+<MethodBody>
+If True Then
+
+Else
+    If False Then
+    |
+    End If
+End If
+</MethodBody>, "Else")
+        End Sub
+
+        <Fact>
+        <Trait(Traits.Feature, Traits.Features.KeywordRecommending)>
+        Public Sub ElseWithinElseIfWithinElse()
+            VerifyRecommendationsContain(
+<MethodBody>
+If True Then
+
+Else
+    If False Then
+    ElseIf True
+        |
+    End If
+End If
+</MethodBody>, "Else")
+        End Sub
+
+        <Fact>
+<Trait(Traits.Feature, Traits.Features.KeywordRecommending)>
+        Public Sub ElseNotWithinDo()
+            VerifyRecommendationsMissing(
+<MethodBody>
+If True Then
+Do While True
```

Comment:
Doesn't actually matter, but can you indent the Do While loop so it's clear nothing weird is going on?